### PR TITLE
Update program

### PR DESCRIPTION
**Pull request by SergeiPetrovAkvelon**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -135,10 +138,14 @@ private BaseObject ChoosePoints(string title, Plant[] menuItems)
 
         public void Tell()
         {
-            plant = (Plant)ChoosePoints("Choose plant:", availablePlants);
+            plant = ChoosePoint<Plant>("Choose plant:", availablePlants);
             Console.Clear();
+            mainCharacter = (Human)ChoosePoint<Character>("Choose main character:", Array.FindAll(availableCharacters, c => c.canPlanting));
```
Comment (tutukin) [inline Program.cs]: It is possible to simply check the type in the predicate: `c => c.GetType() == typeof(Human)`

This way we can drop `canPlanting` field from the Character class.

File: Program.cs [hunk 2]
```diff
@@ -15,7 +16,7 @@ static void Main(string[] args)
 
     class Fairytale
```
Comment (tutukin) [inline Program.cs]: Please, try to follow the rule: one type - one file.

File: Interfaces/IToCall.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Turnip.Classes;
+
+namespace Turnip.Interfaces
```
Comment (tutukin) [inline Interfaces/IToCall.cs:8]: Why do we need these interfaces?

File: Program.cs [hunk 4]
```diff
@@ -147,86 +154,46 @@ public void Tell()
 
         private void WriteFairytaleToConsole()
         {
-            int currentCharacter = 0;
-            Console.WriteLine("{0} planted a {1} and the {1} grew big, very big.", characters[currentCharacter], plant);
-            Console.WriteLine("The {0} began to pull the {1} out of the ground: he pulled and pulled, but could not pull it out.", characters[currentCharacter], plant);
+            activeCharacters++;
+            mainCharacter.ToPlant(plant);
```
Comment (tutukin) [inline Program.cs]: The code below looks a bit too complicated to me.

would it be possible to represent the characters as an array, starting with the main character, then plant and grow, then use a single loop to pull and call?

File: Interfaces/IToCall.cs (hunk 3 above)
Comment (SergeiPetrovAkvelon) [reply Interfaces/IToCall.cs:8]: This interface implements the behavior when a character calls another character.
image

File: Program.cs (hunk 1 above)
Comment (SergeiPetrovAkvelon) [reply Program.cs]: Droped `canPlanting`. Added checking the type.

File: Program.cs (hunk 2 above)
Comment (SergeiPetrovAkvelon) [reply Program.cs]: Moved

File: Program.cs (hunk 4 above)
Comment (SergeiPetrovAkvelon) [reply Program.cs]: I changed the logic of the program and made it simpler.

File: Interfaces/IToCall.cs (hunk 3 above)
Comment (tutukin) [reply Interfaces/IToCall.cs:8]: 1. Interface ~~defines~~ declares, class implements
2. Interfaces are not used to represent objects anywhere.

Generally, we use inheritance for closely related classes: Shape <- Rectangle <- Square, while interfaces allow to make an abstraction over non-related objects, e.g. IRepository <- MySqlRepositury / FileSystemRepository / AzureCosmosRepository.

The point of defining an interface is to use it somewhere. For example, to a SetSomeItemState class constructor we pass the IRepository that allows reading and writing items from any implementation, and allows the business case to not worry about the actual implementation at all.

**Pull request merged**